### PR TITLE
feat(table): add option to control display count of avatars (#416)

### LINKED ISSUE #416: [Table] Add feature to control how many avatars should be shown in `STableCellAvatars`
Currently, `TableCellAvatars` are fixed limit only 3 avatars to be shown. There use case where we want show more users like 10 of them.

Let's add option to specify how many avatars should be shown.

```ts
export interface TableCellAvatars<V = any, R = any> extends TableCellBase {
  type: 'avatars'
  avatars: TableCellAvatarsOption[] | ((value: V, record: R) => TableCellAvatarsOption[])
  color?: 'neutral' | 'soft' | 'mute'

  // @default 2
  avatarCount?: number
  nameCount?: number
}
```

When count is set to 2. It should show 2 avatars, and when there is more than 2 avatars, show placeholder avatar as 3rd avatar (this is new change). Same goes for higher numbers.

For names, when it is set to 2, show 2 names, and then suffix with `+1`.

### PR DESCRIPTION
**Pull request by kiaking**
- close #416

Now we can control how many avatars do we wanna show on table cell. We can also set count to `0` to hide avatars or names.

```ts
const table = useTable({
  columns: {
    authors: {
      label: 'Authors',
      cell: (authors) => ({
        type: 'avatars',
        avatars: authors,
        avatarCount: 5,
        nameCount: 3
      })
    }
  }
})
```

Screenshot 2023-12-19 at 16 35 43

### CODE REVIEW

File: stories/components/STable.01_Playground.story.vue [hunk 1]
```diff
@@ -367,6 +402,7 @@ function updateTagsFilter(value: string) {
 <style scoped>
 .table :deep(.col-name)      { --table-col-width: 160px; }
 .table :deep(.col-status)    { --table-col-width: 144px; }
+.table :deep(.col-authors)   { --table-col-width: 192px; }
```
Comment (brc-dd) [inline stories/components/STable.01_Playground.story.vue:405]:
```suggestion
.table :deep(.col-authors)   { --table-col-width: 240px; }
```

**Review by brc-dd — APPROVED** [review]

**Comment by brc-dd** [comment]
LGTM. The empty box is looking weird though 👀

**Review by NozomuIkuta — APPROVED** [review]
LGTM.

By the way, I agree with @brc-dd's suggestion.

**Comment by kiaking** [comment]
I think this would work. Looks a bit like it got trimmed... but 🤔  What do you think,

Screenshot 2023-12-19 at 17 32 29

**Comment by brc-dd** [comment]
yeah looks better I guess

**Comment by brc-dd** [comment]
We can play around with margins if you want it more compact. For example, this is what github's look like (-11px margin for 20px img):

image

**Comment by kiaking** [comment]
Haha, yeah this is tough one. Also GitHub avatars keep squashing as numbers grow. Somewhat interesting way to handle the "+1" part I think 👀

I guess we can keep it as-is. I think it is more important to show the full avatars so that we can recognize who those are 😳

**Comment by brc-dd** [comment]
sounds good.

**Pull request merged**